modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java

```diff
@@ -214,9 +220,56 @@ public String getHelp() {
 
     @Override
     public void addOperationToGroup(String tag, String resourcePath, Operation operation, CodegenOperation co, Map<String, List<CodegenOperation>> operations) {
-        super.addOperationToGroup(tag, resourcePath, operation, co, operations);
-        if (MICROPROFILE.equals(getLibrary())) {
-            co.subresourceOperation = !co.path.isEmpty();
+        if (!useTags) {
+            String basePath = resourcePath;
```

Comment:
can you please add comment explaining what this code block does, ideally with an example?